### ISSUE TITLE
ipfs tests are broken

### ISSUE THREAD

**Issue by flokli** [NixOS/nixpkgs#74000]
Currently, the ipfs tests don't pass:
```
getter: running command: ipfs --api /ip4/127.0.0.1/tcp/5001 id
getter# Error: cannot connect to the api. Is the deamon running? To run as a standalone CLI command remove the api file in `$IPFS_PATH/api`
getter: exit status 1
(0.05 seconds)
getter: running command: ipfs --api /ip4/127.0.0.1/tcp/5001 id
getter# Error: cannot connect to the api. Is the deamon running? To run as a standalone CLI command remove the api file in `$IPFS_PATH/api`
getter: exit status 1
(0.04 seconds)
getter# [  164.578906] ipfs[2307]: Swarm listening on /ip4/10.0.2.15/tcp/4001
getter# [  164.582481] ipfs[2307]: Swarm listening on /ip4/127.0.0.1/tcp/4001
getter# [  164.586616] ipfs[2307]: Swarm listening on /ip4/192.168.1.2/tcp/4001
getter# [  164.590553] ipfs[2307]: Swarm listening on /ip6/::1/tcp/4001
getter# [  164.593915] ipfs[2307]: Swarm listening on /ip6/fec0::5054:ff:fe12:3456/tcp/4001
getter# [  164.597171] ipfs[2307]: Swarm listening on /ip6/fec0::9935:124d:d4f8:92e6/tcp/4001
getter# [  164.599888] ipfs[2307]: Swarm listening on /p2p-circuit
getter# [  164.612342] ipfs[2307]: Swarm announcing /ip4/10.0.2.15/tcp/4001
getter# [  164.617381] ipfs[2307]: Swarm announcing /ip4/127.0.0.1/tcp/4001
getter# [  164.621227] ipfs[2307]: Swarm announcing /ip4/192.168.1.2/tcp/4001
getter# [  164.624117] ipfs[2307]: Swarm announcing /ip6/::1/tcp/4001
getter# [  164.633730] ipfs[2307]: Swarm announcing /ip6/fec0::5054:ff:fe12:3456/tcp/4001
getter# [  164.638885] ipfs[2307]: Swarm announcing /ip6/fec0::9935:124d:d4f8:92e6/tcp/4001
getter# [  164.644472] ipfs[2307]: API server listening on /ip4/127.0.0.1/tcp/5001
getter# [  164.646929] ipfs[2307]: WebUI: http://127.0.0.1:5001/webui
getter# [  164.648814] ipfs[2307]: 23:45:55.344 ERROR  fuse/ipns: looking up /ipns/QmVS8VuFSicadbGat8n9R34r9KPnNwUhY4fEkTjHRzuv9R: tried GetValue from nil routing ipns_unix.go:107
getter# [  164.652362] ipfs[2307]: 23:45:55.365 ERROR       node: error mounting: tried GetValue from nil routing mount_unix.go:95
getter# [  164.655664] ipfs[2307]: 23:45:55.389 ERROR     swarm2: swarm listener accept error: accept tcp6 [::]:4001: use of closed network connection swarm_listen.go:78
getter# [  164.659235] ipfs[2307]: Error: tried GetValue from nil routing
getter# [  164.661445] systemd[1]: ipfs.mount: Succeeded.
getter# [  164.663532] systemd[1]: ipfs-norouting.service: Main process exited, code=exited, status=1/FAILURE
getter# [  164.665958] systemd[1]: ipfs-norouting.service: Failed with result 'exit-code'.
getter# [  164.667897] systemd[1]: ipfs-norouting.service: Consumed 297ms CPU time, received 0B IP traffic, sent 2.5K IP traffic.
```

cc @fpletz @elitak

**Comment by elitak** [comment]
What do I run to reproduce this? `nix-build -A ipfs` on master works for me.

**Comment by flokli** [comment]
nix-build -A nixosTests.ipfs

**Comment by flokli** [comment]
@elitak were you able to reproduce?

**Comment by elitak** [comment]
No. `error: attribute 'ipfs' in selection path 'nixosTests.ipfs' not found`. This is on latest master, b7c19e1b91. I don't think there's much of a chance I will fix the problem anyway; upstream has very distasteful coding practices that create lots of extra work downstream, here. Should we just remove the tests? or is that already done?

**Comment by flokli** [comment]
Right - it's not added to `all-tests.nix` yet (but should be, if it gets fixed).
Until then, `nix-build nixos/tests/ipfs.nix` should work too.

I don't know about upstreams coding practices, but I assume the tests did once work.

There's no point in shipping broken tests in nixpkgs. If we can't get them to work, we should drop them. In general, tests help maintainers to verify things are still working, or at least not broken in very basic ways.

adding @rvolosatovs and @veprbl because of https://github.com/NixOS/nixpkgs/pull/68159, any opinions?

**Comment by veprbl** [comment]
FWIW I tested 0bf5907aa9494bb341ce65fa526e7b831ae26f10~ and 0bf5907aa9494bb341ce65fa526e7b831ae26f10 and the issue is also seen in 0.4.21 with the same errors.
I'm not an expert on ipfs, but it appears that the issue is not with the package itself. Perhaps the `defaultMode = "norouting";` conflicts with `autoMount = true;` here.

cc @mguentner as the author of the test

**Comment by Mic92** [comment]
I fixed tests in 8b5707b5471c3cda6afb096341b06d835949b6f6
